write-rustdoc-hide-lines/src/code_block_definition.rs

```diff
@@ -0,0 +1,252 @@
+use std::ops::Range;
+
+use crate::hidden_ranges::HiddenRanges;
+
+#[derive(Debug, PartialEq)]
+enum Annotation {
+    HideLines(HiddenRanges),
+    Other(String),
+}
+
+impl From<&str> for Annotation {
+    fn from(text: &str) -> Self {
+        let is_hide_lines = text.starts_with("hide_lines=");
+
+        if is_hide_lines {
+            Annotation::HideLines(
+                text.get(11..)
+                    .unwrap_or("")
+                    .split(' ')
+                    .filter(|r| r.trim() != "")
+                    .map(|range| {
+                        let is_range = range.contains('-');
+
+                        if is_range {
+                            let range = range.split('-').collect::<Vec<_>>();
+                            let start = range[0].parse().unwrap();
+                            let end = range[1].parse().unwrap();
+
+                            Range { start, end }
+                        } else {
+                            let line_no = range.parse::<usize>().unwrap();
+                            Range {
+                                start: line_no,
+                                end: line_no,
+                            }
+                        }
+                    })
+                    .collect(),
+            )
+        } else {
+            Annotation::Other(String::from(text))
+        }
+    }
+}
+
+impl Annotation {
+    fn into_string(&self) -> String {
```

Comment:
Btw @IceSentry, any suggestion for this? I don't understand why it's OK to use the `&self` signature for `CodeBlockDefinition::into_string` though… By clippy, both should be consuming `self` (or that's what I understand): https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

<img width="849" alt="image" src="https://user-images.githubusercontent.com/188612/181756034-289b9d6d-98c8-4882-89e7-9a67ab7d6f50.png">

Reply:
Fixed in 6244271.